Add language metadata best practices from string-meta [I18N-ACTION-1241]

aphillips opened the pull request:
- Add the note text from gpuweb, which is action 1241
- Add the language related best practices from string meta to the "TBD" section provided previously


<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/aphillips/bp-i18n-specdev/pull/106.html" title="Last updated on Jul 13, 2023, 3:06 PM UTC (61c111d)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/bp-i18n-specdev/106/3cef306...aphillips:61c111d.html" title="Last updated on Jul 13, 2023, 3:06 PM UTC (61c111d)">Diff</a>

aphillips committed:
Add language metadata best practices from string-meta [I18N-ACTION-1241]
- Add the note text from gpuweb, which is action 1241
- Add the language related best practices from string meta to the "TBD"
  section provided previously

xfq commented on index.html:
```diff
@@ -471,9 +471,81 @@ <h4>Establishing the language of inline runs</h4>
+</div>
+
+<div class="req" id="bp_language_indexing">
+<p class="advisement">Specifications SHOULD recommend the use of language indexing when localizable strings can be supplied in multiple languages for the same value.</p>
```
I don't quite understand this sentence. Could you provide a simple example?

aphillips replied:
Good callout. See if ec6a09b fixes it?

xfq replied:
Looks good to me. Thanks!

aphillips committed:
Address @xfq's comment by adding text about language indexing
- Add paragraph about localization
   - Add `dfn` for language indexing
   - Add link to String-Meta

aphillips committed:
Fix egregious typo (with a partial rewrite)

aphillips committed:
Address comments from 2023-06-15 teleconference
- Add explanation text to the language fields
- Make id's consistently use underscores

xfq commented on index.html:
```diff
@@ -475,26 +475,42 @@ <h3>Identifying the language of strings</h3>
+<aside class="links" id="lang_strings_links">
+<p class="links_title">Useful background and overviews for this section</p>
+<ul>
+<li class="w3"><p class="link"><a href="https://w3c.github.io/string-meta/">Strings on the Web: Language and Direction Metadata</a>.</p>
```
We should link to the /TR version (same for the links below).

aphillips replied:
Done (globally) for string-meta. Other links possibly need attention, but I'll create a separate issue and PR for those.

aphillips committed:
Update styles to match tr-design
- Add comments to help keep these in sync

Note that we want to override the styles in this repo in order to try
different styles out in the future.

Note that Chrome has a bug in rendering all-small-caps.

aphillips committed:
Address @xfq comment
- Replace w3c.github.io/string-meta with TR link (globally)

aphillips committed:
Merge branch 'w3c:gh-pages' into gh-pages

aphillips committed:
Changes requested in 2023-07-13 telecon

- still awaiting the link-to-stylesheet bits

aphillips commented:
Per teleconference, merging this. I will address the stylesheet linking in a separate pull request.
